packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,328 @@
+ * @export
+ * @class
+ */
+export class LocalFile {
+    tempPath: string;
+    fileName: string;
+    fileType: EditFileType;
+    guiAvail: boolean;
+    zosResp: IZosFilesResponse;
+}
+
+/**
+ * A shared utility class that uss and ds handlers use for local file editing
+ * @export
+ * @class
+ */
+export class EditUtilities {
+    /**
+     * Builds a temp path where local file will be saved. If uss file, file name will be hashed
+     * to prevent any conflicts with file naming. A given filename will always result in the
+     * same unique file path.
+     * @param {LocalFile} lfFile - object containing pertinent information about the local file during the editing process
+     * @returns {Promise<string>} - returns unique file path for temp file
+     * @memberof EditUtilities
+     */
+    public static async buildTempPath(lfFile: LocalFile, commandParameters: IHandlerParameters): Promise<string>{
+        const ext = commandParameters.arguments.extension ?? "txt";
+        if (lfFile.fileType == 'uss'){
+            // Hash in a repeatable way if uss fileName (incase there are special characters in name)
+            const crypto = require("crypto");
+            const hash = crypto.createHash('sha256').update(lfFile.fileName).digest('hex');
+            return tmpdir() +"\\" + hash  + "." + ext;
+        }
+        return tmpdir() + "\\" + lfFile.fileName + "." + ext;
+    }
+
+    /**
+     * Check for temp path's existance (check if previously `stashed` edits exist)
+     * @param {string} tempPath - unique file path for local file (stash/temp file)
+     * @returns {Promise<boolean>} - promise that resolves to true if stash exists or false if doesn't
+     * @memberof EditUtilities
+     */
+    public static async checkForStash(tempPath: string): Promise<boolean>{
+        try {
+            return existsSync(tempPath);
+        } catch(err) {
+            throw new ImperativeError({
+                msg: `Failure when checking for stash. Command terminated.`,
+                causeErrors: err
+            });
+        }
+    }
+
+    /**
+     * Collection of prompts to be used at different points in editing process
+     * @param {Prompt} prompt - selected prompt from {@link Prompt} (enum object)
+     * @param {string} tempPath - unique file path for local file (stash/temp file)
+     * @returns {Promise<boolean>} - promise that resolves depending on prompt case and user input
+     * @memberof EditUtilities
+     */
+    public static async promptUser(prompt: Prompt, tempPath?: string): Promise<boolean>{
+        let input;
+        switch (prompt){
+            case Prompt.useStash:
+                input = await CliUtils.readPrompt(TextUtils.chalk.green(`Keep and continue editing found stash? Y/n`));
+                if (input === null) {
+                    throw new ImperativeError({
+                        msg: `No input provided. Command terminated.`
```

Comment:
Should this message be wrapped in `TextUtils.chalk.red` like the other error messages below?

I'm kind on the fence about whether forcing error messages to be in red text is a good idea 🙂 Typically we already output errors in red so making the message red here too may be a bit redundant.